substrate/frame/election-provider-support/src/lib.rs

```diff
@@ -361,72 +404,103 @@ pub trait ElectionDataProvider {
 	/// Clear all voters and targets.
 	#[cfg(any(feature = "runtime-benchmarks", test))]
 	fn clear() {}
+
+	#[cfg(any(feature = "runtime-benchmarks", test))]
+	fn set_desired_targets(_count: u32) {}
 }
 
-/// Base trait for types that can provide election
-pub trait ElectionProviderBase {
-	/// The account identifier type.
+/// An [`ElectionDataProvider`] that exposes for an external entity to request lock/unlock on
+/// updates in the election data.
+///
+/// This functionality is useful when requesting multi-pages of election data, so that the data
+/// provided across pages (and potentially across blocks) is consistent.
+pub trait LockableElectionDataProvider: ElectionDataProvider {
```

Comment:
You could also always standardize it as: 

calling `electable_targets(p)` if `p > 0` implies `lock`
calling `electing_voters(p)` if `p == 0` implies `unlock`

Reply:
That's a good point. I would prefer to make the lock/unlock explicit for readability and safety. We already have a few implicit assumptions across the pallets that are important to keep in mind and ensure (e.g. `fn elect` should be called from msp to lsp. This would add another important implicit assumption in the staking pallet. 

Happy to be reasoned against adding this extra trait though.

Reply:
I decided to remove the `LockableElectionDataProvider` trait. This implies that we need to call `SortedListProvider::lock`/`unlock` against the voter/target list when appropriated. This will also require the `SortedeListProvider` trait and bags list to support pausing the implicit and explicit rebaging/put in front of. For this PR, we don't need this yet, so I opened a new issue for now.  

Issue here https://github.com/paritytech/polkadot-sdk/issues/6328 will add it to an umbrella ticket for the tasks that need to be completed before rolling out the multi-block election.